test/common.js

```diff
@@ -0,0 +1,19 @@
+var yaml = require('js-yaml');
+
+module.exports.getDiag = function (body) {
+    var yamlStart = body.indexOf('  ---');
+    var yamlEnd = body.indexOf('  ...\n');
+    var diag = body.slice(yamlStart, yamlEnd).split('\n').map(function (line) {
+        return line.slice(2);
+    }).join('\n');
+
+    // The stack trace will vary depending on where the code is run, so just
+    // strip it out.
+    var withStack = yaml.safeLoad(diag);
+    delete withStack.stack;
+    return withStack;
+}
+
+module.exports.stripFullStack = function (output) {
+    return output.replace(/^\W+at.*:\d+:\d+.*$\W/gm, '');
```

Comment:
How about I modify this to something like `verifyAndStripTraces`, which strips all stacks from the generated YAML and checks:

1) That the expected test file shows up somewhere in the stack trace
2) That every line of the stack trace "looks right". Honestly, the only thing I can think of here is matching against some basic regex (or regexes, which might make things more understandable).

For more clarification, here's an example error, with the stacks still fully in the test output:

```
    +++ found                                                                                 
    --- wanted                                                                                
         actual: |-                                                                           
           {}                                                                                 
         stack: |-                                                                            
           Error: should be equivalent                                                        
    +          at Test.assert [as _assert] (/Users/fmurphy/src/tape/lib/test.js:216:54)       
    +          at Test.bound [as _assert] (/Users/fmurphy/src/tape/lib/test.js:65:32)         
    +          at Test.deepEqual.Test.deepEquals.Test.isEquivalent.Test.same (/Users/fmurphy/src/tape/lib/test.js:384:10)
    +          at Test.bound [as deepEqual] (/Users/fmurphy/src/tape/lib/test.js:65:32)       
    +          at Test.<anonymous> (/Users/fmurphy/src/tape/test/undef.js:36:11)              
    +          at Test.bound [as _cb] (/Users/fmurphy/src/tape/lib/test.js:65:32)             
    +          at Test.run (/Users/fmurphy/src/tape/lib/test.js:84:10)                        
    +          at Test.bound [as run] (/Users/fmurphy/src/tape/lib/test.js:65:32)             
    +          at Immediate.next (/Users/fmurphy/src/tape/lib/results.js:71:15)               
    +          at runCallback (timers.js:574:20)                                              
    +          at tryOnImmediate (timers.js:554:5)                                            
    +          at processImmediate [as _immediateCallback] (timers.js:533:5)                  
       ...  
```

If any refactoring adds or removes a function, pulls something into an anonymous function, or moves things into a new file ... then this stack trace will change. Even if we do obvious things, like strip line numbers and character positions, and normalize file paths.

Completely checking stack correctness would require sophisticated code analysis, which seems like overkill. The challenge here is thus figuring out the right tradeoff between "smart" parsing and effort involved.